modules/fedimint-ln-client/src/lib.rs

```diff
@@ -125,6 +152,28 @@ pub enum LnReceiveState {
     Claimed,
 }
 
+async fn check_internal_payment(
+    instance: &ClientModuleInstance,
+    invoice: &Invoice,
+    gateway: &LightningGateway,
+) -> FederationResult<bool> {
+    // Asserts that the first hop of the invoice route hints
+    // leads into the current federation
+    //
+    // FIXME: Account for multiple gateways serving the same federation
+    let check_first_hop_to_current_fed = invoice
+        .route_hints()
+        .first()
+        .and_then(|rh| rh.0.last())
+        .map(|hop| (hop.src_node_id, hop.short_channel_id))
+        == Some((gateway.node_pub_key, gateway.mint_channel_id));
+
+    Ok(
+        check_first_hop_to_current_fed
+            && instance.api.offer_exists(*invoice.payment_hash()).await?,
```

Comment:
Not sure if we want to check whether the offer exists right here. If the route hint checks out, but no offer exists, then I think it will currently try to pay via the gateway. But the first step of the internal payment state machine could check this and fail the payment without needing to connect to the gateway. 

I think either one would probably work, just thinking out loud.